src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1744 @@
+// src/Beman/Optional26/tests/optional_constexpr.t.cpp                -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <Beman/Optional26/optional.hpp>
+
+#include <functional>
+#include <ranges>
+#include <tuple>
+
+#include <gtest/gtest.h>
+
+TEST(OptionalConstexprTest, TestGTest) { ASSERT_EQ(1, 1); }
+
+namespace {
+struct empty {};
+struct no_default {
+    no_default()                             = delete;
+    no_default(const no_default&)            = default;
+    no_default(no_default&&)                 = default;
+    no_default& operator=(const no_default&) = default;
+    no_default& operator=(no_default&&)      = default;
+    no_default(empty) {};
+};
+
+struct base {
+    int i_;
+    constexpr base() : i_(0) {}
+    constexpr base(int i) : i_(i) {}
+};
+
+struct derived : public base {
+    int j_;
+    constexpr derived() : base(0), j_(0) {}
+    constexpr derived(int i, int j) : base(i), j_(j) {}
+};
+} // namespace
+
+TEST(OptionalConstexprTest, Constructors) {
```

Comment:
Can we reduce the duplication between optional.t.cpp and optional_constexpr.t.cpp?